### PR TITLE
refactor: use destination config from config-be than from message

### PR DESCRIPTION
**Pull request by koladilip**
# Description
Earlier transformer was sending destConfig in message and it contains secrets so it is appearing in live events so refactored code to use config from CP.

## Notion Ticket

[Lambda Keys exposed in Live events](https://www.notion.so/rudderstacks/CFD-Destination-config-leaking-to-Live-Events-for-AWS-Lambda-0efdfbd385384462b0f194dca2c66bfd)

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: services/streammanager/lambda/lambdamanager.go [hunk 1]
```diff
@@ -15,14 +16,13 @@ import (
...
 type destinationConfig struct {
-	InvocationType string
-	ClientContext  string
-	Lambda         string
+	InvocationType string `json:"invocationType"`
+	ClientContext  string `json:"clientContext"`
+	Lambda         string `json:"lambda"`
 }
 
 type inputConfig struct {
```
Comment (saikumarrs) [inline services/streammanager/lambda/lambdamanager.go:24]: The name `inputConfig`, is it still relevant?

File: services/streammanager/lambda/lambdamanager.go [hunk 2]
```diff
@@ -71,9 +71,14 @@ func (producer *LambdaProducer) Produce(jsonData json.RawMessage, _ interface{})
...
-	if config == nil {
-		return 400, "Failure", "[Lambda] error :: Invalid destination config"
+	var config destinationConfig
+	err = mapstructure.Decode(destConfig, &config)
+	if err != nil {
+		returnMessage := "[Lambda] error while unmarshalling destConfig :: " + err.Error()
+		return 400, "Failure", returnMessage
+	}
+	if config.InvocationType == "" {
+		config.InvocationType = "Event"
```
Comment (saikumarrs) [inline services/streammanager/lambda/lambdamanager.go:81]: Is this change relevant? Also, is there a test case for this?

Comment (koladilip) [reply services/streammanager/lambda/lambdamanager.go:81]: Yes this required as this not part of the destConfig, we are hard coding it.

Comment (utsabc) [reply services/streammanager/lambda/lambdamanager.go:81]: We are always sending this from transformer there won't be any scenario of this being empty string

File: services/streammanager/lambda/lambdamanager_test.go [hunk 3]
```diff
@@ -81,26 +82,6 @@ func TestProduceWithInvalidData(t *testing.T) {
...
-	// Invalid Destination Config
-	sampleDestConfig := map[string]interface{}{}
-	sampleEventJson, _ = json.Marshal(map[string]interface{}{
-		"payload":    sampleMessage,
-		"destConfig": "invalid dest config",
-	})
-	statusCode, statusMsg, respMsg = producer.Produce(sampleEventJson, sampleDestConfig)
-	assert.Equal(t, 400, statusCode)
-	assert.Equal(t, "Failure", statusMsg)
-	assert.Contains(t, respMsg, "[Lambda] error while unmarshalling jsonData")
```
Comment (utsabc) [inline services/streammanager/lambda/lambdamanager_test.go:103]: Why are we removing these tests?

File: services/streammanager/lambda/lambdamanager.go (hunk 1 above)
Comment (koladilip) [reply services/streammanager/lambda/lambdamanager.go:24]: yes, it is to hold what transformer is sending right?

File: services/streammanager/lambda/lambdamanager.go (hunk 2 above)
Comment (saikumarrs) [reply services/streammanager/lambda/lambdamanager.go:81]: @utsabc the plan is to not use the config sent from the transformer.

Comment (koladilip) [reply services/streammanager/lambda/lambdamanager.go:81]: We won't be sending this in the future from transformer so this required here.

Comment (utsabc) [reply services/streammanager/lambda/lambdamanager.go:81]: Got it

File: services/streammanager/lambda/lambdamanager_test.go (hunk 3 above)
Comment (koladilip) [reply services/streammanager/lambda/lambdamanager_test.go:103]: Destination config is not part the input message any more so this is not required anymore

File: services/streammanager/lambda/lambdamanager.go (hunk 1 above)
Comment (saikumarrs) [reply services/streammanager/lambda/lambdamanager.go:24]: It's the input payload, right? Now, we are not even sending the config.

**Review by utsabc — APPROVED** [review]

File: services/streammanager/lambda/lambdamanager.go (hunk 1 above)
Comment (koladilip) [reply services/streammanager/lambda/lambdamanager.go:24]: https://github.com/rudderlabs/rudder-server/blob/master/services/streammanager/common/common.go#L16
we are expecting json as input in the method.
https://github.com/rudderlabs/rudder-transformer/blob/master/src/v0/destinations/lambda/transform.js#L66
We are sending this structure from transformer.

Comment (saikumarrs) [reply services/streammanager/lambda/lambdamanager.go:24]: I'm just talking about the name `inputConfig`. Should it be something like `inputData` at least?

Comment (koladilip) [reply services/streammanager/lambda/lambdamanager.go:24]: Sorry I missed reading the name part. inputData is fine

Comment (koladilip) [reply services/streammanager/lambda/lambdamanager.go]: Updated

**Review by sandeepdsvs — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**